ogr/ogrsf_frmts/hana/ogrhanadatasource.cpp

```diff
@@ -623,8 +623,9 @@ int OGRHanaDataSource::Open(const char *newName, char **openOptions, int update)
         {
             odbc::DatabaseMetaDataRef dbmd = conn_->getDatabaseMetaData();
```

Comment:
Since you already have a function that determines HANA version, it makes sense to move everything related to versions in it and rename the function. Then we don't have HANA version logic spread across different places.

```
void OGRHanaDataSource::DetermineVersions()
{
    odbc::DatabaseMetaDataRef dbmd = conn_->getDatabaseMetaData();
    CPLString dbVersion(dbmd->getDBMSVersion());
    hanaVersion_ = HANAVersion::fromVersionString(dbVersion);

    if (hanaVersion_.major() < 4)
    {
        cloudVersion_ = HANAVersion(0, 0, 0);
        return;
    }

    odbc::StatementRef stmt = conn_->createStatement();
    const char *sql = "SELECT CLOUD_VERSION FROM SYS.M_DATABASE;";

    odbc::ResultSetRef rsVersion = stmt->executeQuery(sql);
    if (rsVersion->next())
        cloudVersion_ =
            HANAVersion::fromVersionString(rsVersion->getString(1)->c_str());

    rsVersion->close();
}
```

Reply:
Done.